core/clustersmngr/client.go

```diff
@@ -0,0 +1,168 @@
+package clustersmngr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	ErrClusterNotFound error = errors.New("cluster not found")
```

Comment:
can we turn this into an error type that lets us pass in the cluster name? might be useful for debugging